### PR TITLE
Enhance/update admin settings UI

### PR DESCRIPTION
**Pull request by MdAsifHossainNadim**
### All Submissions:
- Update dokan admin settings page ui.
- Update admin delivery time settings and added 24 hours picking option for admin.
- Added delivery times upgrader for admin and vendor delivery time settings.

* [ ] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [ ] My code satisfies feature requirements
* [ ] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

### Changes proposed in this Pull Request:

Closes # .
Related with: https://github.com/weDevsOfficial/dokan-pro/pull/1851.

### How to test the changes in this Pull Request:

1. Go to "Dokan -> Settings".
2. And checked all the admin settings field.
3. Go to "Dokan -> Settings -> Delivery Time".
4. Checked admin delivery times.

### Changelog entry

> Update dokan admin settings ui.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [x] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [x] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

### CODE REVIEW

File: includes/Admin/Settings.php [hunk 1]
```diff
@@ -484,13 +511,13 @@ public function get_settings_fields() {
             ),
             'dokan_withdraw'   => [
                 'withdraw_methods'    => [
-                    'name'    => 'withdraw_methods',
-                    'label'   => __( 'Withdraw Methods', 'dokan-lite' ),
-                    'desc'    => __( 'Select suitable Withdraw methods for Vendors', 'dokan-lite' ),
-                    'type'    => 'multicheck',
-                    'default' => [ 'paypal' => 'paypal' ],
-                    'options' => dokan_withdraw_get_methods(),
-                    'tooltip' => __( 'Check to add available payment methods for vendors to withdraw money.', 'dokan-lite' ),
```
Comment (tanmayjay) [inline includes/Admin/Settings.php:493]: Why these tooltips are removed?

File: src/admin/components/Fields.vue [hunk 2]
```diff
@@ -1,316 +1,338 @@
...
-
-                <input
-                    :type="fieldData.type || 'text'"
-                    class="regular-text"
-                    :class="[ { 'dokan-input-validation-error': hasValidationError( fieldData.name ) }, fieldData.class ]"
-                    :id="sectionId + '[' + fieldData.name + ']'"
-                    :name="sectionId + '[' + fieldData.name + ']'"
-                    v-model="fieldValue[fieldData.name]"
-                >
+            <div class="field_contents" v-bind:class="[fieldData.content_class ? fieldData.content_class : '']">
```
Comment (tanmayjay) [inline src/admin/components/Fields.vue:13]: Make sure both the static and binded classes are included after compilation.

File: src/admin/components/Fields.vue [hunk 3]
```diff
@@ -1,316 +1,338 @@
...
+                                    <option v-for="( optionVal, optionKey ) in fieldData.options.week" :value="optionKey" v-html="optionVal"></option>
+                                </select>
+                            </div>
+                        </div>
+                        <div class="col-3">
+                            <div class="dokan-input-group">
+                                <span class="dokan-input-group-addon">{{ __( 'Day', 'dokan-lite' ) }}</span>
+                                <select v-if="!fieldData.grouped" class="regular" :name="sectionId + '[' + fieldData.name + '][days]'" :id="sectionId + '[' + fieldData.name + ']'" v-model="fieldValue[fieldData.name]['days']">
+                                    <option v-if="fieldData.placeholder" value="" v-html="fieldData.placeholder"></option>
+                                    <option v-if="!( 'L' !== fieldValue[fieldData.name]['week'] && ( 'saturday' === optionKey || 'sunday'=== optionKey ) )" v-for="( optionVal, optionKey ) in fieldData.options.days" :value="optionKey" v-html="optionVal"></option>
```
Comment (tanmayjay) [inline src/admin/components/Fields.vue:331]: Never use `v-if` and `v-for` inside same element. Wrap it with a `template` element and use `v-if` inside that.
Apply this everywhere the `v-if` and `v-for` used inside the same element.

File: src/admin/components/Fields.vue [hunk 4]
```diff
@@ -460,185 +479,207 @@
...
+                    <div class="field_data">
+                        <h3 class="field_heading" scope="row">
+                            {{ fieldData.label }}
+                        </h3>
+                        <p class="field_desc" v-html="fieldData.desc"></p>
+                    </div>
+                    <div class="field">
+                        <select v-if="!fieldData.grouped" class="regular" :name="sectionId + '[' + fieldData.name + ']'" :id="sectionId + '[' + fieldData.name + ']'" v-model="fieldValue[fieldData.name]">
+                            <option v-if="fieldData.placeholder" value="" v-html="fieldData.placeholder"></option>
+                            <option v-for="( optionVal, optionKey ) in fieldData.options" :value="optionKey" v-html="optionVal"></option>
```
Comment (tanmayjay) [inline src/admin/components/Fields.vue:497]: While using `v-for`, it's best practice to use a unique key binding. Ensure this for every `v-for`. For example, in this case,
```
<option v-for="( optionVal, optionKey ) in fieldData.options" :key="optionKey" :value="optionKey" v-html="optionVal"></option>
```

File: src/admin/components/Fields.vue [hunk 5]
```diff
@@ -648,45 +689,300 @@
...
                 <p v-if="hasError( fieldData.name )" class="dokan-error">
                     {{ getError( fieldData.label ) }}
                 </p>
-                <p class="description" v-html="fieldData.desc"></p>
-            </td>
+            </div>
         </template>
-    </tr>
+
+        <template v-if="'social' === fieldData.type">
```
Comment (tanmayjay) [inline src/admin/components/Fields.vue:730]: This component is too big. Can it be sliced into multiple components?
Also, here some similar things are being repeated by checking with `v-if`. See if a common component can be built to be used by passing the parameter `fieldData.type`.

**Review by tanmayjay — CHANGES_REQUESTED** [review]

File: includes/Admin/Settings.php (hunk 1 above)
Comment (MdAsifHossainNadim) [reply includes/Admin/Settings.php:493]: Because, tooltip is not managed in our new design...

**Review by tanmayjay — APPROVED** [review]

**Pull request closed without merging**